forage-android/src/test/java/com/joinforage/forage/android/vault/RosettaPinSubmitterTest.kt

```diff
@@ -0,0 +1,276 @@
+        assertEquals(500, firstError.httpStatusCode)
+        assertEquals("unknown_server_error", firstError.code)
+    }
+
+    @Test
+    fun `Rosetta returns a ForageError`() = runTest {
+        val paymentRef = "xyzForageError456"
+        server.givenRosettaPaymentCaptureRequest("xyzForageError456").returnsUnauthorized()
+
+        val result = executeSubmit(paymentRef)
+
+        assertTrue(result is ForageApiResponse.Failure)
+        val firstError = (result as ForageApiResponse.Failure).errors[0]
+        assertEquals(
+            """
+            [forage] Received ForageError from forage: Code: missing_merchant_account
+            Message: No merchant account FNS number was provided.
+            Status Code: 401
+            Error Details (below):
+            null
+            """.trimIndent(),
+            mockLogger.errorLogs.last().getMessage()
+        )
+        assertEquals("No merchant account FNS number was provided.", firstError.message)
+        assertEquals(401, firstError.httpStatusCode)
+        assertEquals("missing_merchant_account", firstError.code)
+    }
+
+    @Test
+    fun `Basis Theory responds with a malformed error`() = runTest {
+        val paymentRef = "xyzMalformedError456"
+        server.givenRosettaPaymentCaptureRequest(paymentRef).returnsMalformedError()
+
+        val result = executeSubmit(paymentRef)
+
+        assertTrue(result is ForageApiResponse.Failure)
+        val firstError = (result as ForageApiResponse.Failure).errors[0]
+        assertEquals(
+            "Failed to send request to Forage Vault.",
+            mockLogger.errorLogs.last().getMessage()
+        )
+        assertEquals("Unknown Server Error", firstError.message)
+        assertEquals(500, firstError.httpStatusCode)
+        assertEquals("unknown_server_error", firstError.code)
+    }
+
+    // The paths dictate which mock server response to expect
+    // The paymentRef is used to determine the path
+    private suspend fun executeSubmit(paymentRef: String): ForageApiResponse<String> {
+        return submitter.submitProxyRequest(buildMockVaultProxyRequest(paymentRef))
+    }
+
+    companion object {
+        private val mockData = MockServiceFactory.ExpectedData
+        private const val MOCK_IDEMPOTENCY_KEY = "idempotency-abcdef123"
+
+        /**
+         * Comes from [MockLogger.getTraceIdValue]
+         */
+        private const val MOCK_TRACE_ID = "11223344"
+        private const val MOCK_ROSETTA_CARD_TOKEN = "tok_rosetta_abcde123"
+
+        private fun buildMockPaymentCapturePath(paymentRef: String) = "/api/payments/$paymentRef/capture/"
+
+        // The paths dictate which mock server response to expect
+        private fun buildMockVaultProxyRequest(paymentRef: String): VaultProxyRequest {
+            val path = buildMockPaymentCapturePath(paymentRef)
+
+            return VaultProxyRequest.emptyRequest()
+                .setHeader(ForageConstants.Headers.X_KEY, "22320ce0-1a3c-4c64-970c-51ed7db34548")
```

Comment:
Do we need to send this `X-KEY` header?

Reply:
We don't need it for Rosetta, but on this line, I'm emulating the headers that are set by the Abstract /Base Vault Submitter